ntbs-service/Pages/Notifications/NotificationsBase.cshtml.cs

```diff
@@ -16,31 +18,87 @@ public abstract class NotificationModelBase : ValidationModel
         }
 
         public Notification Notification { get; set; }
+
         [BindProperty]
         public int? NotificationId { get; set; }
         
         // This can be thrown away once proper banner work completes
         public NotificationStatus NotificationStatus { get; set; }
 
+        [ViewData]
+        public Dictionary<string, List<string>> ErrorDictionary { get; set; }
+
         public async Task<IActionResult> OnPostSubmitAsync()
         {
+            // First Validate and Save current page details
             bool isValid = await ValidateAndSave(NotificationId);
-            if (!isValid) {
+            if (!isValid) 
+            {
                 return await OnGetAsync(NotificationId);
             }
-            
-            var notification = await service.GetNotificationAsync(NotificationId);
-            if (notification == null)
+
+            // Get Notifications with all owned properties to check for 
+            Notification = await service.GetNotificationWithAllInfoAsync(NotificationId);
+            if (Notification == null)
             {
                 return NotFound();
             }
+            
+            // IsRequired fields on Models requires ShouldValidateFull flag
+            SetShouldValidateFull();
+
+            if (!IsModelValid())
+            {
+                GetModelStateErrors();
+                return Partial("./NotificationErrorSummary", this);
+            } 
 
-            await service.SubmitNotification(notification);
+            await service.SubmitNotification(Notification);
             
             return RedirectToPage("../Overview", new {id = NotificationId});
         }
 
-        
+        private bool IsModelValid()
+        {
+            return TryValidateModel(Notification);
+        }
+
+        private void SetShouldValidateFull() 
+        {
+            Notification.ShouldValidateFull = true;
+            Notification.SocialRiskFactors.ShouldValidateFull = true;
+            Notification.PatientDetails.ShouldValidateFull = true;
+            Notification.PatientTBHistory.ShouldValidateFull = true;
+            Notification.ContactTracing.ShouldValidateFull = true;
+            Notification.ClinicalDetails.ShouldValidateFull = true;
+            Notification.Episode.ShouldValidateFull = true;
+            Notification.NotificationSites.ForEach(x => x.Notification = Notification);
+        }
+
+        // This method converts Model State Errors to Dictionary of Errors to be used in view
+        private void GetModelStateErrors() 
+        {
+            ErrorDictionary = new Dictionary<string, List<string>>();
+
+            foreach (var key in ModelState.Keys) 
+            {
+                // Splitting '[' as well due to List properties having index [0]

```

Comment:
Do you mean "Splitting on '['..." ?
Maybe giving an example of a property name here would make it clearer :)